src/png.rs

```diff
@@ -181,13 +189,19 @@ impl From<ColorType> for (png::ColorType, png::BitDepth) {
     fn from(ct: ColorType) -> (png::ColorType, png::BitDepth) {
         use self::png::ColorType::*;
         let (ct, bits) = match ct {
-            ColorType::Gray(bits) => (Grayscale, bits),
-            ColorType::RGB(bits) => (RGB, bits),
-            ColorType::Palette(bits) => (Indexed, bits),
-            ColorType::GrayA(bits) => (GrayscaleAlpha, bits),
-            ColorType::RGBA(bits) => (RGBA, bits),
-            ColorType::BGRA(bits) => (RGBA, bits),
-            ColorType::BGR(bits) => (RGB, bits),
+            ColorType::Unknown(bits) => (Indexed, bits),
```

Comment:
This is not the reverse of the `From` conversion above. Maybe we should avoid a direction one-to-one conversion for now.

Reply:
`From` should *never* panic. But we might want to just not give a `From` implementation for that reverse case anyhow.

Reply:
I've just removed this From implementation and now return an error if an unsupported ColorType is passed to encode. Really that function should return an ImageResult instead of an io::Result but that is a much deeper rabbit hole than I wanted to get into for this PR.